### DISCUSSION TITLE
Allow creating global JS functions to reuse in component design

### DISCUSSION THREAD

**Discussion by micebrain** [Budibase/budibase#4257] [Ideas]
Hello
I created a few calculation formulas in JS for display in the UI - thanks to the JS instead of Handlebars capability. 
But I found myself copying and pasting the same function to each component that requires it i.e. a lot of duplication and if I update the function I have to remember not to miss any component that uses it and risk creating bugs in the app.
It becomes a real pain when the app grows. 
It would be nice to have a place  for example in the "Data" screen a section for defining JS functions that can be made available throughout the app. 
In this way, we avoid duplication and also reduced the final application size.

**Comment by mjashanks** [comment]
Agreed - this would be great!

**Comment by cantonsen** [comment]
Is this something that would be considered done by the Budibase team?
I'd really love to see this feature made available.

**Comment by rochesterj** [comment]


**Reply by cantonsen** [reply]
While true it would probably work, but to me this sounds more like a hack than a workaround 🙃

**Reply by melohagan** [reply]
I think eval can have some security implications as well. There's an issue you can follow here: https://github.com/Budibase/budibase/issues/3571

**Comment by Simoon-F** [comment]
I need it！
